apps/dashboard/src/layout/root.tsx

```diff
@@ -0,0 +1,22 @@
+import { Outlet } from 'react-router-dom';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { AuthProvider, ClerkProvider, SegmentProvider } from '@/context';
+import { HelmetProvider } from 'react-helmet-async';
+
+const queryClient = new QueryClient();
+
+export const Root = () => {
+  return (
+    <QueryClientProvider client={queryClient}>
```

Comment:
Please move these to the root file of the app to have everything in one place. This is not the layout after all; it's all the top-level providers. It's easier to have them in one place in a top level file.

Reply:
This is what they suggest doing in the latest react-router, basically to have the `main.tsx` entry file and define the Root component for your shared code and outlet, [ref](https://reactrouter.com/en/main/start/tutorial).
The thing is that the new `RouterProvider` doesn't take children as a prop, so we can't go with the suggestion of having App and providers defined there as some of them are dependent on the router.

Reply:
You can ignore my comment. In the suggestion they use the name `routes` that I think it's better than layout that speaks about pure UI concerns.